sympy/core/add.py

```diff
@@ -610,10 +612,12 @@ def _eval_is_nonpositive(self):
         from sympy.core.exprtools import _monotonic_sign
         if not self.is_number:
             c, a = self.as_coeff_Add()
-            if not c.is_zero and a.is_nonpositive:
+            if not c.is_zero:
                 v = _monotonic_sign(a)
                 if v is not None:
                     s = v + c
+                    if a.is_nonnegative and v.is_integer == True:
```

Comment:
Can you explain why this change works? I'm unclear what this code actually does (it could use some comments) and why is_integer would come into play here.